src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java

```diff
@@ -76,16 +84,19 @@ public void stop() {
         }
     }
 
-    private void runShardSync() {
+    private TaskResult runShardSync() {
+        TaskResult result = null;
         try {
             if (leaderDecider.isLeader(workerId)) {
                 LOG.debug(String.format("WorkerId %s is a leader, running the shard sync task", workerId));
-                metricsEmittingShardSyncTask.call();
+                result = metricsEmittingShardSyncTask.call();
             } else {
                 LOG.debug(String.format("WorkerId %s is not a leader, not running the shard sync task", workerId));
             }
         } catch (Throwable t) {
             LOG.error("Error during runShardSync.", t);
         }
+
+        return result == null ? new TaskResult(null) : result;
```

Comment:
Why do we not propagate any exception to TaskResult?

Reply:
Any exception that occurs in `metricsEmittingShardSyncTask.call()` will be caught and propagated in `result`, which holds the last exception as a member. The only case where the Throwable catch block would be exercised is if something goes wrong in the logging or the `isLeader` method, neither of which are expected to throw. This was kept mostly for consistency with existing code.

Reply:
If result == null, that means metricsEmittingShardSyncTask.call() was having some issues and didn't return the expected result, right? Why do we return a valid test result with no exception in this case?

Reply:
Removed this code as part of changes in above comment